### PR TITLE
Update download request fetching on topic and topic content pages

### LINKED ISSUE #10564: Remote Content Browsing: Update `useDownloadRequests` to provide remote content download progress (complete/incomplete)
A follow-up to https://github.com/learningequality/kolibri/pull/10558

Related backend issue #10391. Not that for the scope of this issue, we don't need granular progress in percentage yet though, but complete/incomplete information would be sufficient.

After backend is ready to provide remote content download progress information, the following places need to be updated in the `useDownloadRequests` composable that currently contains mocks instead.

**(1)` addDownloadRequest` function saves mocked `complete` boolean after timeout ([reference](https://github.com/learningequality/kolibri/pull/10558/files#diff-b2f2ba0aa5c79f8ff5454440121da8b1c4c52167ad2ea45972a659530c96ff9eR122-R127))**:

```js
requestData.complete = false;
setTimeout(() => (requestData.complete = true), 1000);
```
This needs to be removed and replaced by a real update which doesn't need to necessarily be implemented in `addDownloadRequest`. Also, `complete` field is just a random mock name and can be changed.

**(2) `isDownloadingByLearner` function uses mocked `complete` boolean ([reference](https://github.com/learningequality/kolibri/pull/10558/files#diff-b2f2ba0aa5c79f8ff5454440121da8b1c4c52167ad2ea45972a659530c96ff9eR156-R169))**:

```js
  function isDownloadingByLearner(content) {
    if (!content || !content.id) {
      return false;
    }
    const downloadRequest = downloadRequestMap.downloads[this.content.id];
    return downloadRequest && !downloadRequest.complete;
  }
```
It's currently using mocked `complete` mentioned in (1). We may or may not need to refactor it depending on how progress logic will be implemented. Note that this function is expected to be reactive by components that use it and return accurate information as the progress changes from incomplete to complete.

**(3) `isDownloadedByLearner` function uses mocked `complete` boolean ([reference](https://github.com/learningequality/kolibri/pull/10558/files#diff-b2f2ba0aa5c79f8ff5454440121da8b1c4c52167ad2ea45972a659530c96ff9eR171-R187))**:

```js
  function isDownloadedByLearner(content) {
    if (!content || !content.id) {
      return false;
    }
    const downloadRequest = downloadRequestMap.downloads[this.content.id];
    return downloadRequest && downloadRequest.complete;
  }
```

Same as (2).

## Completion criteria

- [ ] Mocked logic described above is replaced by a real implementation
- [ ] All components relying on `isDownloadingByLearner` and `isDownloadedByLearner` still work as expected or are updated to new implementation in case there are radical changes to the composable
- [ ] Content remote download feature still passes all QA scenarios described in #10558

### LINKED ISSUE #11239: Downloads are not properly loaded on TopicsPage and TopicsContentPage
## Observed behavior
We currently load the first page of download requests

There's no guarantee (if someone has a lot of download requests) that the currently displayed resource(s) will be included in those first 100

## Expected behavior
For the TopicsPage - download filtered by the displayed resource ids.
Also, when doing any 'load more' behaviour, ensure that additional download requests are loaded for the newly fetched resources.

For the TopicsContentPage - download filtered by the displayed resource id.

In both cases, we should show a loading state in place of the add/remove download button until the data has been loaded.

### PR DESCRIPTION
**Pull request by rtibbles**
## Summary
* Makes the ContentRequest endpoint accept contentnode_id and contentnode_ids filtering parameters
* Fetches download requests on the topics page by passing the loaded contentnode_ids and further when doing requests for 'more'
* Fetches just the download request for the specific resource on the content page
* Cleans up reactivity on topic content page when a resource is downloaded
* Fixes an unnoticed bug where the networkDevices map was being filtered on, rather than just being looked up in
* Refactors the MyDownload page polling behaviour into the downloadRequest composable
* Reuses this polling behaviour on the TopicsContentPage to ensure we get timely status updates of individual resource downloads

## References
Fixes [#10564](https://github.com/learningequality/kolibri/issues/10564)
Fixes [#11239](https://github.com/learningequality/kolibri/issues/11239)

## Reviewer guidance
Ensure there are no regressions on the MyDownloads page.
Ensure that Download Requests are properly displayed on the topics page and the topics content page.
Ensure that adding a new download in both places properly updates the frontend in a timely fashion.

----

## Testing checklist

- [ ] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests

## PR process

- [ ] PR has the correct target branch and milestone
- [ ] PR has 'needs review' or 'work-in-progress' label
- [ ] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

## Reviewer checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

### CODE REVIEW

File: kolibri/plugins/learn/assets/src/views/TopicsPage/index.vue [hunk 1]
```diff
@@ -361,14 +361,40 @@
...
       const channel = ref(null);
       const contents = ref([]);
       const loading = ref(true);
 
+      const _getAllDescendantChildren = topic => {
+        const contentnode_id__in = [];
+        const children = topic.children ? topic.children.results : [];
+        for (const child of children) {
+          if (child.kind === ContentNodeKinds.TOPIC) {
+            contentnode_id__in.push(..._getAllDescendantChildren(child));
```
Comment (marcellamaki) [inline kolibri/plugins/learn/assets/src/views/TopicsPage/index.vue:378]: I know recursion isn't actually all that uncommon, but I do enjoy seeing in the wild where it makes complete sense and not just in a documentation page from my bootcamp classes where I was only understanding it theoretically and felt so confused as a junior dev

File: kolibri/plugins/learn/assets/src/views/TopicsPage/index.vue [hunk 2]
```diff
@@ -361,14 +361,40 @@
...
+          } else {
+            contentnode_id__in.push(child.id);
+          }
+        }
+        return contentnode_id__in;
+      };
+
+      const fetchRemoteBrowsingContentNodeUserData = topic => {
+        if (get(isUserLoggedIn) && props.deviceId) {
```
Comment (marcellamaki) [inline kolibri/plugins/learn/assets/src/views/TopicsPage/index.vue:387]: `props.deviceId` is this a way to access the props from within setup?

File: kolibri/plugins/learn/assets/src/views/TopicsPage/index.vue [hunk 3]
```diff
@@ -827,7 +860,7 @@
         const parent = parentIndex > -1 ? this.contents[parentIndex] : null;
         const more = parent && parent.children && parent.children.more;
         if (more) {
-          if (this.isUserLoggedIn) {
+          if (this.isUserLoggedIn && !this.deviceId) {
```
Comment (marcellamaki) [inline kolibri/plugins/learn/assets/src/views/TopicsPage/index.vue:863]: I have seen the existence check on the deviceId in a few files now and I don't really know what it's for

File: kolibri/plugins/learn/assets/src/composables/useDownloadRequests.js [hunk 4]
```diff
@@ -85,11 +81,12 @@ export default function useDownloadRequests(store) {
       metadata,
       source_id: store.getters.currentUserId,
       source_instance_id: get(instanceId),
-      reason: 'UserInitiated',
+      reason: 'USER_INITIATED',
       facility: store.getters.currentFacilityId,
-      status: 'Pending',
+      status: 'PENDING',
```
Comment (marcellamaki) [inline kolibri/plugins/learn/assets/src/composables/useDownloadRequests.js:130]: important changes here

File: kolibri/plugins/learn/assets/test/views/topic-content.spec.js [hunk 5]
```diff
@@ -192,7 +192,12 @@ describe('TopicsContentPage', () => {
         beforeEach(async () => {
           useDownloadRequests.mockImplementation(() =>
             useDownloadRequestsMock({
-              isDownloadedByLearner: () => true,
+              downloadRequestMap: {
+                [CONTENT_ID]: {
+                  id: CONTENT_ID,
+                  status: 'COMPLETED',
```
Comment (marcellamaki) [inline kolibri/plugins/learn/assets/test/views/topic-content.spec.js:198]: this is the sort of thing in a test that I need to remember for some future date, that I can actually put in a test

**Review by marcellamaki — APPROVED** [review]
I had some questions, but no blockers, just learning questions. Thanks @rtibbles !

File: kolibri/plugins/learn/assets/src/views/TopicsPage/index.vue (hunk 2 above)
Comment (rtibbles) [reply kolibri/plugins/learn/assets/src/views/TopicsPage/index.vue:387]: This is _the_ way to access it.

Importantly, the `props` object is reactive, but if you destructure `const { deviceId } = props` then the `deviceId` will not be reactive.

File: kolibri/plugins/learn/assets/src/views/TopicsPage/index.vue (hunk 3 above)
Comment (rtibbles) [reply kolibri/plugins/learn/assets/src/views/TopicsPage/index.vue:863]: If it's not defined then we're not browsing remotely. If it is, we are.

**Review by pcenov — APPROVED** [review]
LGTM, no regression issues observed as well.

**Pull request merged**